src/topics/create.ts

```diff
@@ -0,0 +1,380 @@
+    ]);
+
+    if (locked && !isAdminOrMod) {
+        throw new Error('[[error:topic-locked]]');
+    }
+
+    if (!scheduled && deleted && !isAdminOrMod) {
+        throw new Error('[[error:topic-deleted]]');
+    }
+
+    if (scheduled && !canSchedule) {
+        throw new Error('[[error:no-privileges]]');
+    }
+
+    if (!canReply) {
+        throw new Error('[[error:no-privileges]]');
+    }
+}
+
+export = function (Topics: TopicFields) {
+    async function onNewPost(postData: PostObjectPartial, data: TopicData) {
+        const { tid } = postData;
+        const { uid } = postData;
+        await Topics.markAsUnreadForAll(tid);
+        await Topics.markAsRead([tid], uid);
+        const [
+            userInfo,
+            topicInfo,
+        ]: (UserObjectSlim | TopicObject)[] = await Promise.all([
+            // The next line calls a function in a module that has not been updated to TS yet
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+            posts.getUserInfoForPosts([postData.uid], uid) as UserObjectSlim[],
+            Topics.getTopicFields(tid, ['tid', 'uid', 'title', 'slug', 'cid', 'postcount', 'mainPid', 'scheduled']),
+            Topics.addParentPosts([postData]),
+            Topics.syncBacklinks(postData),
+            // The next line calls a function in a module that has not been updated to TS yet
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+            posts.parsePost(postData),
+        ]) as (UserObjectSlim | TopicObject)[];
+
+        postData.user = userInfo[0] as UserObjectSlim;
+        postData.topic = topicInfo as TopicObject;
+        postData.index = (topicInfo.postcount as unknown) as number - 1;
+        // The next line calls a function in a module that has not been updated to TS yet
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+        posts.overrideGuestHandle(postData, data.handle);
+
+        postData.votes = 0;
+        postData.bookmarked = false;
+        postData.display_edit_tools = true;
+        postData.display_delete_tools = true;
+        postData.display_moderator_tools = true;
+        postData.display_move_tools = true;
+        postData.selfPost = false;
+        postData.timestampISO = utils.toISOString(postData.timestamp) as string;
+        postData.topic.title = String(postData.topic.title);
+
+        return postData;
+    }
+
+    Topics.create = async function (data: TopicData) {
+        // This is an internal method, consider using Topics.post instead
+        const timestamp = data.timestamp || Date.now();
+
+        // The next line calls a function in a module that has not been updated to TS yet
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
```

Comment:
nit: same as above comment as privileges; just need to make sure whoever is in charge of the db stuff knows to delete these lines